src/ik.rs

```diff
@@ -215,39 +215,71 @@ where
         constraints: &Constraints,
     ) -> Result<DVector<T>, Error> {
         let operational_space = define_operational_space(&constraints);
+        let required_dof = operational_space.iter().filter(|x| **x).count();
         let orig_positions = arm.joint_positions();
-        let dof = orig_positions.len();
+        let available_dof = arm.dof() - constraints.ignored_joint_names.len();
+
         let t_n = arm.end_transform();
         let err = calc_pose_diff_with_constraints(target_pose, &t_n, operational_space);
-        let orig_positions = arm.joint_positions();
         let mut jacobi = jacobian(arm);
-        let use_dof = operational_space.iter().filter(|x| **x).count();
-        let mut removed_count = 0;
+        let mut num_removed_rows = 0;
         for (i, use_i) in operational_space.iter().enumerate() {
             if !use_i {
-                jacobi = jacobi.remove_row(i - removed_count);
-                removed_count += 1;
+                jacobi = jacobi.remove_row(i - num_removed_rows);
+                num_removed_rows += 1;
+            }
+        }
+
+        let mut ignored_joint_indices = Vec::<usize>::new();
+        for joint_name in constraints.ignored_joint_names.iter() {
+            // Try to get joint index
+            match arm.iter_joints().position(|x| x.name == *joint_name) {
```

Comment:
Finding string from Vec<String> is slow. I want to this logic to out of the loop.
If you can change the interface of `solve_one_loop_with_constraints`, is it possible to move out this logic?